### PR TITLE
Add PermissionController:hasAllPermissions

### PR DESCRIPTION
**Pull request by FrederikBolding**
Add `hasAllPermissions` function to `PermissionController`.

**Description**
- ADDED:
  - Add `hasAllPermissions` function to `PermissionController`.
    - For use in `snaps-skunkworks` to check for multiple permissions in one call

**Checklist**

- [x] Tests are included if applicable
- [x] Any added code is fully documented

### CODE REVIEW

File: src/permissions/PermissionController.ts [hunk 1]
```diff
@@ -845,6 +860,24 @@ export class PermissionController<
     return Boolean(this.getPermission(origin, target));
   }
 
+  /**
+   * Checks whether the subject with the specified origin has all the specified
+   * permissions.
+   *
+   * @param origin - The origin of the subject.
+   * @param targets - The target names of every permission.
+   * @returns Whether the subject has all the permissions.
```
Comment (Gudahtt) [inline src/permissions/PermissionController.ts:869]: Nit: This sounds like it's suggesting that `targets` _must_ be a list of all permissions. But this is just the list of permissions being checked, right? And similarly the return description could use the same term as the method description.

```suggestion
   * @param targets - The list of permissions to check.
   * @returns Whether the subject has all the specified permissions.
```

File: src/permissions/PermissionController.test.ts [hunk 2]
```diff
@@ -4384,6 +4384,76 @@ describe('PermissionController', () => {
       expect(hasPermissionsSpy).toHaveBeenNthCalledWith(2, 'foo');
     });
 
+    it('action: PermissionController:hasAllPermissions', () => {
```
Comment (Gudahtt) [inline src/permissions/PermissionController.test.ts:4387]: I was expecting to see both the public method and the action tested. The coverage threshold doesn't demand it, but that's just a guideline. The public method is part of the public API of this package, so we want to ensure its behavior is fully tested so we don't break it accidentally and unknowingly ship a breaking change.

Comment (Gudahtt) [reply src/permissions/PermissionController.test.ts:4387]: That would be a good opportunity to break this into smaller tests as well. Right now it's not clear what this test is trying to prove, there is too much going on. The case of calling `hasAllPermissions` when the subject has none, some, or all of the target permissions could be three separate test cases. Then the test description can make clear exactly what's being tested.

Probably worth adding a test case for calling `hasAllPermissions` with an empty array as well. And for when the subject has additional permissions.

**Comment by Gudahtt** [comment]
What was the reason for adding this method? If there's a use case you already had in mind, that would be helpful to mention here.

**Comment by FrederikBolding** [comment]
> What was the reason for adding this method? If there's a use case you already had in mind, that would be helpful to mention here.

My bad for not linking the issue. It's a potential solution to a snaps problem here: https://github.com/MetaMask/snaps-skunkworks/issues/348 - Still waiting for some feedback, so potentially I was a bit too fast in marking this as ready for review 😄

**Comment by FrederikBolding** [comment]
Closing in favor of a different solution for now.
